test/Microsoft.SqlTools.ServiceLayer.Test/QueryExecution/ExecutionPlanTests.cs

```diff
@@ -0,0 +1,251 @@
+        {
+            // Setup:
+            // ... I have a batch that has been executed with a execution plan 
+            Batch b = Common.GetExecutedBatchWithExecutionPlan();
+
+            // If:
+            // ... I have a result set and I ask for a valid execution plan
+            ResultSet planResultSet = b.ResultSets.First();
+            ExecutionPlan plan = planResultSet.GetExecutionPlan().Result;
+
+            // Then:
+            // ... I should get the execution plan back
+            Assert.Equal("xml", plan.Format);
+            Assert.Contains("Execution Plan", plan.Content);
+        }
+
+        [Fact]
+        public void ExecutionPlanInvalid()
+        {
+            // Setup:
+            // ... I have a batch that has been executed
+            Batch b = Common.GetBasicExecutedBatch();
+
+            // If:
+            // ... I have a result set and I ask for an execution plan that doesn't exist
+            ResultSet planResultSet = b.ResultSets.First();
+            ExecutionPlan plan = planResultSet.GetExecutionPlan().Result;
+
+            // Then:
+            // ... I should get a null value since it has no execution plan 
+            Assert.Null(plan);
+        }
+
+        #endregion
+
+        #region Batch Class Tests
+
+        [Fact]
+        public void BatchExecutionPlanValidTest()
+        {
+            // If I have an executed batch which has an execution plan 
+            Batch b = Common.GetExecutedBatchWithExecutionPlan();
+
+            // ... And I ask for a valid execution plan 
+            ExecutionPlan plan = b.GetExecutionPlan(0).Result;
+
+            // Then:
+            // ... I should get the execution plan back
+            Assert.Equal("xml", plan.Format);
+            Assert.Contains("Execution Plan", plan.Content);
+        }
+
+        [Fact]
+        public void BatchExecutionPlanInvalidTest()
+        {
+            // Setup:
+            // ... I have a batch that has been executed without an execution plan 
+            Batch b = Common.GetBasicExecutedBatch();
+
+            // If: 
+            // ... I ask for an invalid execution plan 
+            ExecutionPlan plan = b.GetExecutionPlan(0).Result;
+
+            // Then:
+            // ... I should get a null value since it has no execution plan 
+            Assert.Null(plan);
+        }
+
+        [Theory]
+        [InlineData(-1)]  // Invalid result set, too low
+        [InlineData(2)]   // Invalid result set, too high
+        public void BatchExecutionPlanInvalidParamsTest(int resultSetIndex)
+        {
+            // If I have an executed batch which has an execution plan 
+            Batch b = Common.GetExecutedBatchWithExecutionPlan();
+
+            // ... And I ask for an execution plan with an invalid result set index
+            // Then: 
+            // ... It should throw an exception
+            Assert.ThrowsAsync<ArgumentOutOfRangeException>(() => b.GetExecutionPlan(resultSetIndex)).Wait();
```

Comment:
One cool thing we get from XUnit is the ability to have async test methods. Although it's not a big deal, since the code isn't very different, we can avoid using the .Wait and instead using the await keyword.